### ISSUE TITLE
Better cachability of message endpoint fragments

### ISSUE THREAD

**Issue by adamghill** [adamghill/django-unicorn#115]
Hotwire returns a 304 HTTP status code (https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304) if the DOM content didn't change.

Maybe add a new `unicorn:hash` which is just a shortuuid of the MD5 hash of the current HTML (does the MD5 hash need to happen?) or just the first 8 of the MD5?, but maybe `unicorn:checksum` could be utilized.

**Comment by adamghill** [comment]
This is implemented in https://github.com/adamghill/django-unicorn/pull/162.

**Comment by adamghill** [comment]
`checksum` couldn't be used because it only verifies the data, not the rendered content. I ended up added a new `shortuuid`, `hash`, that gets set when the component is initialized and then re-generated on every AJAX update. If the generated hash is the same as the old hash in `views.py`, a 304 is returned which is checked in `messageSender.js` and short-circuits any other processing.

Closing this because #162 is merged into master and should be released in 0.23.0.
